internal/restorer/filerestorer.go

```diff
@@ -327,25 +333,33 @@ func (r *fileRestorer) loadBlob(rd io.Reader, blobID restic.ID, length int) ([]b
 		return nil, errors.Errorf("blob %v returned invalid hash", blobID)
 	}
 
-	return plaintext, nil
+	// move decrypted data to the start of the buffer
+	copy(buf, plaintext)
```

Comment:
It would be possible to avoid the data relocation by returning both `plaintext` and `buf`.

Reply:
Good idea! is changed.